### PR TITLE
ddt removal in sync plan and test_multiple_paths tests

### PR DESCRIPTION
**Pull request by sthirugn**
- Also optimized the usage of test data in test_multiple_paths.py by creating single entities list and excluding necessary entities as required.

### CODE REVIEW

**Comment by sthirugn** [comment]
Test results:

test_syncplan.py:
```sh
Finding files... done.
Importing test modules ... done.

----------------------------------------------------------------------
Ran 12 tests in 348.312s

OK (skipped=1)
```

test_multiple_paths.py local tests are running now.

File: tests/foreman/api/test_syncplan.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# pylint: disable=invalid-name
```
Comment (sthirugn) [inline tests/foreman/api/test_syncplan.py]: this is added because the test names are too long.  I wanted to shorten the names but some folks are interested in keeping the long/descriptive names.  This is debatable.  

But for now, this workaround is to silence pylint.

File: tests/foreman/api/test_multiple_paths.py [hunk 2]
```diff
@@ -247,52 +205,31 @@ def test_post_status_code(self, entity_cls):
...
+                logger.debug('test_post_status_code arg: %s', entity_cls)
+                skip_if_sam(self, entity_cls)
+
+                # Libvirt compute resources suffer from BZ 1118015. However,
+                # partials cannot be compared for class identity and the class
+                # hierarchy needs fixing (SatelliteQE/nailgun#42), so we just
+                # comment it out above.
+                if (entity_cls in BZ_1118015_ENTITIES and
+                        bz_bug_is_open(1118015)):
+                    self.skipTest('Bugzilla bug 1118015 is open.')
```
Comment (omaciel) [inline tests/foreman/api/test_multiple_paths.py]: I'm curious to know what happens if this condition is true? Will the rest of the test be skipped once it is true?

**Comment by omaciel** [comment]
ACK

File: tests/foreman/api/test_multiple_paths.py (hunk 2 above)
Comment (sthirugn) [inline tests/foreman/api/test_multiple_paths.py]: The rest of the test will be run normally. 
Demo test:
```py
    def test_subtest_demo(self):
        i = 0
        for item in range(4):
            with self.subTest(item):
                if item is not 1:
                    print 'Running test %d' % item
                    i += 1
                else:
                    unittest2.SkipTest('I cannot execute value == 1 ')
                    print 'Skipping test %d' % item
                    i += 1
        print 'Total number of tests executed is %d' % i
```

Output:
```sh
nosetests --nocapture tests/foreman/cli/test_org_dupe.py
Running test 0
Skipping test 1
Running test 2
Running test 3
Total number of tests executed is 4
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK

# python -m unittest2 tests/foreman/cli/test_org_dupe.py
Running test 0
Skipping test 1
Running test 2
Running test 3
Total number of tests executed is 4
.
----------------------------------------------------------------------
Ran 1 test in 0.001s

OK
```

Obviously we need https://github.com/xmlrunner/unittest-xml-reporting/issues/85 to make jenkins aware of the skipped ones.

File: tests/foreman/api/test_multiple_paths.py [hunk 3]
```diff
@@ -153,49 +149,33 @@ def test_get_status_code(self, entity_cls):
...
-        entities.Repository,
-        entities.Role,
-        entities.Subnet,
-        entities.System,
-        entities.TemplateKind,
-        entities.User,
-        entities.UserGroup,
-    )
-    def test_get_unauthorized(self, entity_cls):
+        for entity_cls in set(valid_entities()) - set(exclude_list):
```
Comment (JacobCallahan) [inline tests/foreman/api/test_multiple_paths.py]: :+1:

File: tests/foreman/api/test_multiple_paths.py [hunk 4]
```diff
@@ -402,58 +286,37 @@ def test_put_status_code(self, entity_cls):
...
+
+                # Create an entity
+                entity_id = entity_cls().create_json()['id']
+
+                # Update that entity.
+                entity = entity_cls()
+                entity.create_missing()
+                response = client.put(
+                    entity_cls(id=entity_id).path(),
+                    # FIXME: use entity.update_payload()
```
Comment (JacobCallahan) [inline tests/foreman/api/test_multiple_paths.py]: is this fixme something that can be accomplished?

Comment (sthirugn) [inline tests/foreman/api/test_multiple_paths.py]: This is out of scope for this PR as this needs time for investigation.

File: tests/foreman/api/test_syncplan.py (hunk 1 above)
Comment (JacobCallahan) [inline tests/foreman/api/test_syncplan.py]: i am in favor of shortening names to common abbreviations and context-based word removal. So ```test_create_sync_plan_with_random_name``` could be changed to ```create_with_rand_name```. This also strips test from the name which could go either way.

File: tests/foreman/api/test_multiple_paths.py (hunk 4 above)
Comment (JacobCallahan) [inline tests/foreman/api/test_multiple_paths.py]: that is what i thought :)

**Comment by JacobCallahan** [comment]
ACK

File: tests/foreman/api/test_syncplan.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/api/test_syncplan.py]: yeah the problem is that `test_` is a mandatory word.  So all test would have a mandatory 5 letters text (`test_`) which leaves around 26 characters to fit everything else.  But yeah this is doable.

Comment (JacobCallahan) [inline tests/foreman/api/test_syncplan.py]: ahhh got it

Comment (sthirugn) [inline tests/foreman/api/test_syncplan.py]: I am updating the PR now :)

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769
